include/swift/AST/Module.h

```diff
@@ -830,6 +830,9 @@ class SourceFile final : public FileUnit {
   /// May be -1, to indicate no association with a buffer.
   int BufferID;
 
+  /// \brief Whether this file is sourced from a primary input.
+  bool IsPrimaryInput;
```

Comment:
This part (and the general indication of which SourceFiles are primary) ought to be run by @davidungar, he's been trying to centralize reasoning about primary-ness and it'd be a shame if this worked against his efforts.

Reply:
I don't yet understand why this is needed. There is a test in FrontendInputsAndOutputs called isInputPrimary. It would be more consistent to use that, but I have just started looking at this PR.
Tx for pinging me, @graydon.

Reply:
Yes, please do not put information about primary-ness into SourceFile. It's the responsibility of clients to not try to do things to non-primary SourceFiles, and most of the compiler is set up so that that can't happen by accident.

Reply:
Thanks for your feedback. I wanted to add an isPrimary predicate to SourceFile because I thought I needed some way to get that information at the SIL level. As Jordan and Slava pointed out, there should be a simpler approach.